openverse_catalog/dags/providers/provider_api_scripts/cleveland_museum.py

```diff
@@ -1,97 +1,57 @@
-            batch = response_json["data"]
-            image_count = _handle_response(batch)
-            logger.info(f"Total images till now {image_count}")
-            offset += LIMIT
+class ClevelandDataIngester(ProviderDataIngester):
+    def __init__(
+        self,
+        providers={"image": prov.CLEVELAND_DEFAULT_PROVIDER},
+        endpoint="http://openaccess-api.clevelandart.org/api/artworks/",
+        delay=5,
+        batch_limit=1000,
+        retries=3,
+        headers: Optional[Dict] = {},
+    ):
+        # Initialize the DataIngester with appropriate values
+        super().__init__(providers, endpoint, delay, batch_limit, retries, headers)
+
+    def get_next_query_params(self, old_query_params, **kwargs):
+        if not old_query_params:
+            # Return default query params on the first request
+            return {"cc": "1", "has_image": "1", "limit": self.batch_limit, "skip": 0}
         else:
-            logger.error("No more images to process")
-            logger.info("Exiting")
-            condition = False
-    image_count = image_store.commit()
-    logger.info(f"Total number of images received {image_count}")
-
-
-def _build_query_param(offset=0, default_query_param=None):
-    if default_query_param is None:
-        default_query_param = DEFAULT_QUERY_PARAMS
-    query_param = default_query_param.copy()
-    query_param.update(skip=offset)
-    return query_param
-
-
-def _get_response(query_param, endpoint=ENDPOINT, retries=RETRIES):
-    response_json, total_images, tries = None, 0, 0
-    for tries in range(retries):
-        response = delay_request.get(endpoint, query_param)
-        if response is not None and response.status_code == 200:
-            try:
-                response_json = response.json()
-                total_images = len(response_json["data"])
-            except Exception as e:
-                logger.warning(f"response not captured due to {e}")
-                response_json = None
-            if response_json is not None and total_images is not None:
-                break
-
-        logger.info(
-            "Retrying \n"
-            f"endpoint -- {endpoint} \t"
-            f" with parameters -- {query_param} "
-        )
-    if tries == retries - 1 and ((response_json is None) or (total_images is None)):
-        logger.warning("No more tries remaining. Returning Nonetypes.")
-        return None, 0
-    else:
-        return response_json, total_images
-
-
-def _handle_response(batch):
-    total_images = 0
-    for data in batch:
-        license_ = data.get("share_license_status", "").lower()
+            # Increment `skip` by the batch limit.
+            return {
+                **old_query_params,
+                "skip": old_query_params["skip"] + self.batch_limit,
+            }
+
+    def get_media_type(self, record):
+        # This provider only supports Images.
+        return "image"
+
+    def get_record_data(self, record):
```

Comment:
I seriously love this!! This is going to make the patterns so much easier to follow 🤩